lib/uring/uring.ml

```diff
@@ -136,16 +129,15 @@ let submit t =
 let errno_is_retry = function -62 | -11 | -4 -> true |_ -> false
 
 let fn_on_ring fn t =
-   let id, res = fn t.uring in
-   match id, res with
+   let (id : Heap.ptr), res = fn t.uring in
+   match (id :> int), res with
    | -1, res when errno_is_retry res ->
```

Comment:
How can `id` (a `Heap.ptr`) be `-1`?

Reply:
Yes, this is a hack around the C-stubs wanting to use `-1` as an error code and me wanting to keep IDs `private`. We could just drop the `private` (and change some assertions to errors in `heap.ml`). What do you think?